lib/kitchen/driver/openstack.rb

```diff
@@ -117,7 +117,7 @@ def openstack_server
           provider: 'OpenStack'
         }
         required_server_settings.each { |s| server_def[s] = config[s] }
-        optional_server_settings.each { |s| server_def[s] = config[s] }
+        optional_server_settings.each { |s| server_def[s] = config[s] if config[s] }
```

Comment:
This check prevents a few failed expectations with exact objects that don't contain keys with nil values. Fog shouldn't care if we have a key with a nil value, but we don't really need to include one anyway.

```
rspec ./spec/kitchen/driver/openstack_spec.rb:316 # Kitchen::Driver::Openstack#openstack_server returns a hash of server settings
rspec ./spec/kitchen/driver/openstack_spec.rb:353 # Kitchen::Driver::Openstack#compute all requirements provided creates a new compute connection
rspec ./spec/kitchen/driver/openstack_spec.rb:359 # Kitchen::Driver::Openstack#compute all requirements provided creates a new network connection
```